crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4567,6 +4567,23 @@ impl<'db> TypeInferenceBuilder<'db> {
                 let param_type = self.infer_type_expression(parameters);
                 UnionType::from_elements(self.db, [param_type, Type::none(self.db)])
             }
+            KnownInstanceType::Union => {
+                let mut builder = UnionBuilder::new(self.db);
+                match parameters {
+                    ast::Expr::Name(_) => {
+                        builder = builder.add(self.infer_type_expression(parameters))
```

Comment:
Dummy question: Is it required to use the builder when we infer a single type? Can't we just return the type itself?

Reply:
It has some extra logic to merge unions for example. I just looked and I can use the `UnionType::from_elements` since it does build the builder: https://github.com/Glyphack/ruff/blob/04f7fc4220591b2a15c011706d056fc7e6629ebe/crates/red_knot_python_semantic/src/types.rs#L2833

Reply:
That makes sense. I'm just surprised that creating an union from one element would result in a different return type than just that type (shouldn't `infer_type_expression` refine that type already?). Looking at the `UnionBuilder::add`, it seems that `UnionBuilder::default().add(ty).build() == ty`

Reply:
Thanks I understand your point now. I removed creating union from a single element.